pkg/clients/coredata/event.go

```diff
@@ -72,426 +68,86 @@ func (e *EventRestClient) init(params types.EndpointParams) {
 	}
 }
 
-// Helper method to decode an event slice
-func (e *EventRestClient) decodeEventSlice(resp *http.Response) ([]models.Event, error) {
-	eSlice := make([]models.Event, 0)
-	dec := json.NewDecoder(resp.Body)
-	err := dec.Decode(&eSlice)
+// Helper method to request and decode an event slice
+func (e *EventRestClient) requestEventSlice(url string) ([]models.Event, error) {
+	data, err := getRequest(url)
 	if err != nil {
-		fmt.Println(err)
+		return []models.Event{}, err
 	}
 
+	eSlice := make([]models.Event, 0)
+	err = json.Unmarshal(data, &eSlice)
 	return eSlice, err
 }
 
-// Helper method to decode an event and return the event
-func (e *EventRestClient) decodeEvent(resp *http.Response) (models.Event, error) {
-	dec := json.NewDecoder(resp.Body)
-	ev := models.Event{}
-	err := dec.Decode(&ev)
+// Helper method to request and decode an event and return the event
+func (e *EventRestClient) requestEvent(url string) (models.Event, error) {
+	data, err := getRequest(url)
 	if err != nil {
-		fmt.Println(err)
+		return models.Event{}, err
 	}
 
+	ev := models.Event{}
+	err = json.Unmarshal(data, &ev)
 	return ev, err
 }
 
 // Get a list of all events
 func (e *EventRestClient) Events() ([]models.Event, error) {
-	req, err := http.NewRequest(http.MethodGet, e.url, nil)
-	if err != nil {
-		fmt.Println(err.Error())
-		return []models.Event{}, err
-	}
-
-	resp, err := makeRequest(req)
-	if err != nil {
-		fmt.Println(err.Error())
-		return []models.Event{}, err
-	}
-	if resp == nil {
-		fmt.Println(ErrResponseNil)
-		return []models.Event{}, ErrResponseNil
-	}
-	defer resp.Body.Close()
-
-	if resp.StatusCode != http.StatusOK {
-		bodyBytes, err := getBody(resp)
-		if err != nil {
-			fmt.Println(err.Error())
-			return []models.Event{}, err
-		}
-		return []models.Event{}, types.NewErrServiceClient(resp.StatusCode, bodyBytes)
-	}
-
-	return e.decodeEventSlice(resp)
+	return e.requestEventSlice(e.url)
 }
 
 // Get the event by id
 func (e *EventRestClient) Event(id string) (models.Event, error) {
-	req, err := http.NewRequest(http.MethodGet, e.url+"/"+id, nil)
-	if err != nil {
-		fmt.Println(err)
-		return models.Event{}, err
-	}
-
-	resp, err := makeRequest(req)
-	if err != nil {
-		fmt.Println(err.Error())
-		return models.Event{}, err
-	}
-	if resp == nil {
-		fmt.Println(ErrResponseNil)
-		return models.Event{}, ErrResponseNil
-	}
-	defer resp.Body.Close()
-
-	if resp.StatusCode != http.StatusOK {
-		bodyBytes, err := getBody(resp)
-		if err != nil {
-			fmt.Println(err.Error())
-			return models.Event{}, err
-		}
-
-		return models.Event{}, types.NewErrServiceClient(resp.StatusCode, bodyBytes)
-	}
-
-	return e.decodeEvent(resp)
+	return e.requestEvent(e.url + "/" + id)
```

Comment:
I do not like this sting constant hardcoding. Also, I find `fmt.Sprintf()` more readable here.